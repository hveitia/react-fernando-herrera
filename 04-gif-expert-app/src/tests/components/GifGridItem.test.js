import React from 'react'
const { shallow } = require("enzyme")
const { GifGridItem } = require("../../components/GifGridItem")

describe('Test on <GifGridItem/>', () => {
    
    const url = '';
    const title = '';
    const wrapper = shallow(<GifGridItem url={url} title={title}/>);

    test('Should be render the component.', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Should have one paragraph with tittle.', () => {      
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('Should have one image with source and alt.', () => {   
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Should have animate__animated class', () => {      
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__animated')).toBe(true);
    })
        
})
