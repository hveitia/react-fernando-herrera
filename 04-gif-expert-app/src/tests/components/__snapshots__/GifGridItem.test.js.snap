// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem/> Should be render the component. 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt=""
    src=""
  />
  <p />
</div>
`;
